src/app/api/swarm/stakgraph/services/route.ts

```diff
@@ -48,38 +204,111 @@ export async function GET(request: NextRequest) {
-      swarmUrl: `https://${getSwarmVanityAddress(swarm.name)}:3355`,
-      endpoint: "/services",
-      method: "GET",
-      params: {
-        ...(clone === "true" ? { clone } : {}),
-        ...(repo_url ? { repo_url } : {}),
-        ...(githubProfile?.username ? { username: githubProfile?.username } : {}),
-        ...(githubProfile?.pat ? { pat: githubProfile?.pat } : {}),
-      },
-      apiKey: encryptionService.decryptField("swarmApiKey", swarm.swarmApiKey),
-    });
+    let responseData: { services: ServiceConfig[] };
+    let containerFiles: Record<string, string> | undefined;
+
+    if (agent === "true" && repo_url) {
+      // Agent mode - call services_agent endpoint
+      try {
+        const { owner, repo } = parseGithubOwnerRepo(repo_url);
 
-    // Transform response format: wrap array in {services: [...]} for new format
-    const responseData = Array.isArray(apiResult.data)
-      ? { services: apiResult.data as ServiceConfig[] }
-      : (apiResult.data as { services: ServiceConfig[] });
+        // Start the agent request
+        const agentInitResult = await swarmApiRequestAuth({
+          swarmUrl: `https://${swarmVanityAddress}:3355`,
+          endpoint: "/services_agent",
+          method: "GET",
+          params: { owner, repo },
+          apiKey: decryptedApiKey,
+        });
+
+        if (!agentInitResult.ok) {
+          console.error("Agent init failed:", agentInitResult);
+          throw new Error("Failed to initiate agent");
+        }
+
+        const initData = agentInitResult.data as { request_id: string };
+        if (!initData.request_id) {
+          throw new Error("No request_id received from agent");
+        }
+
+        // Poll for completion
+        const agentResult = await pollAgentProgress(
+          swarmVanityAddress,
+          initData.request_id,
+          decryptedApiKey
+        );
+
+        if (!agentResult.ok) {
+          throw new Error("Agent failed to complete");
+        }
+
+        const agentFiles = agentResult.data as Record<string, string>;
+
+        // Parse pm2.config.js to extract services
+        const services = agentFiles["pm2.config.js"]
+          ? parsePM2ConfigToServices(agentFiles["pm2.config.js"])
+          : [];
+
+        responseData = { services };
+        containerFiles = agentFiles;
+
+      } catch (error) {
+        console.error("Agent mode failed, falling back to standard mode:", error);
+        // Fall back to standard services endpoint
+        const apiResult = await swarmApiRequestAuth({
+          swarmUrl: `https://${swarmVanityAddress}:3355`,
+          endpoint: "/services",
+          method: "GET",
+          params: {
+            ...(clone === "true" ? { clone } : {}),
+            ...(repo_url ? { repo_url } : {}),
+            ...(githubProfile?.username ? { username: githubProfile?.username } : {}),
+            ...(githubProfile?.pat ? { pat: githubProfile?.pat } : {}),
```

Comment:
use githubProfile.appAccessToken || githubProfile.pat

`appAccessToken` is from the Github app. We should always use it if its available. But having a fallback is good in case we decide to support no-github-app (for public repo not in org)